### DISCUSSION TITLE
Generic user-defined models (not necessarily described by conventional DAEs)

### DISCUSSION THREAD

**Discussion by diodezener** [CURENT/andes#155] [Modeling] [answered]
Dr. Cui,

I'm wondering is there a way to write generic user-defined models (as maybe voltage controlled current source) instead of a set of traditional DAEs. For instance, if I want to represent a dynamic load by some kind of equivalence (possibly via data-driven methods), is there a way to incorporate it into the existing framework?

Thanks,

Dz

**Comment by cuihantao** [comment, marked as answer]
Hi @diodezener , welcome!
The user-defined models in ANDES is generic "power injection" model, as opposed to current injection models used in commercial software (where sources are modeled as voltage behind impedance). Power injection model should allow you to easily write a model as long as you compute the P and Q and impose it onto the linked buses.

For voltage-controlled current source, you can compute the P and Q for the network and implement internal logic for the current. 

If you have more specific descriptions, I'm glad to provide a direction for implementation.
